flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/TemporalTableJoinTest.scala

```diff
@@ -156,12 +156,12 @@ class TemporalTableJoinTest extends TableTestBase {
       inputRates: TemporalTableFunction,
       proctime: Boolean = false): Unit = {
     val rates = inputRates.asInstanceOf[TemporalTableFunctionImpl]
-    assertTrue(rates.getPrimaryKey.isInstanceOf[ResolvedFieldReference])
-    assertEquals("currency", rates.getPrimaryKey.asInstanceOf[ResolvedFieldReference].name)
-    assertTrue(rates.getTimeAttribute.isInstanceOf[ResolvedFieldReference])
+    assertTrue(rates.getPrimaryKey.isInstanceOf[FieldReferenceExpression])
```

Comment:
nit: use object equality instead of casting. Just construct the expression that you except.